### PR TITLE
Productization_US01: Productize branding

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**

### CODE REVIEW

File: portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue [hunk 1]
```diff
@@ -14,42 +14,60 @@
...
       <div class="logoForm boxContent">
-        <h4>{{ $t('companyName.label') }}</h4>
+        <h4 class="labelStyle">{{ $t('companyName.label') }}</h4>
         <div>
-          <input id="companyNameInput" v-model="branding.companyName" :placeholder="$t('companyName.placeholder')" type="text" name="formOp" class="ignore-vuetify-classes" value="">
-          <div class="info">{{ $t('companyName.input.hint') }}</div>
-        </div> 
-        <h4>
+          <div class="info"><p>{{ $t('companyName.input.hint') }}</p></div>
+          <input id="companyNameInput" v-model="branding.companyName" :placeholder="$t('companyName.placeholder')" type="text" name="formOp" style="margin-top: 4px;margin-left: -9px;width: 32%;" class="ignore-vuetify-classes inputCompany" value="">
```
Comment (azayati) [inline portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue]: don't add style directly

File: portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue [hunk 2]
```diff
@@ -14,42 +14,60 @@
...
           {{ $t('selectlogo.label') }}
         </h4>
+        <div class="info labelLocation"><p>{{ $t('noteselectlogo.label') }}</p></div>
         <div class="clearfix">
-          <div class="pull-left">
+          <div v-show="showLogo" class="pull-left">
+            <div id="previewLogo" class="previewLogo">
+              <div class="btnAction">
+                <i v-if="removeLogoButtonDisplayed" :title="$t('branding.message.remove.logo')" class="fas fa-trash-alt iconDelete" rel="tooltip" data-placement="bottom" @click="removeLogo"></i>
+                <i v-if="!removeLogoButtonDisplayed" :title="$t('branding.message.change.logo')" class="uiIconEdit iconEdit" rel="tooltip" data-placement="bottom" @click="editLogo"></i>
```
Comment (azayati) [inline portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue]: v-else instead

File: portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue [hunk 3]
```diff
@@ -14,42 +14,60 @@
...
+              </div>
+              <div>
+                <img id="previewLogoImg" :src="logoPreview" alt="" class="previewImg">
```
Comment (azayati) [inline portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue]: empty alt not needed ?

File: portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue [hunk 4]
```diff
@@ -14,42 +14,60 @@
...
+            </div>
+          </div>
+          <div v-show="!showLogo" class="pull-left">
             <div class="fileDrop">
               <div ref="dropFileBox" class="dropZone">
-                <label class="dropMsg" for="attachLogo">
-                  <i class="uiIcon attachFileIcon"></i> {{ $t('attachment') }}
-                </label>
+                <div>
+                  <div class="firstBlock">
```
Comment (azayati) [inline portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue]: privilege significant name for the classes

File: portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue [hunk 5]
```diff
@@ -14,42 +14,60 @@
...
           </div>
         </div>
+        <hr class="lastSeparation">
         <div class="themeStyle boxContent">
           <h4>
             {{ $t('themeColors.label') }}
           </h4>
 
-          <v-row>
+          <v-row class="blockColors">
```
Comment (azayati) [inline portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue]:
```suggestion
          <v-row class="colorsBlock">
```

File: portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue [hunk 6]
```diff
@@ -141,6 +136,8 @@ export default {
     }
   },
   created() {
+    console.log(brandingConstants.HOMEICON);
```
Comment (azayati) [inline portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue]: remove console.log

File: portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue [hunk 7]
```diff
@@ -167,11 +164,8 @@ export default {
     }.bind(this));
   },
   methods: {
-    convertImageDataAsSrc(imageData) {
-      let binary = '';
-      const bytes = new Uint8Array(imageData);
-      bytes.forEach(byte => binary += String.fromCharCode(byte));
-      return `data:image/png;base64,${btoa(binary)}`;
+    editLogo(){
```
Comment (azayati) [inline portlet/exoadmin/src/main/webapp/company-branding-app/components/ExoCompanyBranding.vue]: you can put this directly in the template

File: portlet/exoadmin/src/main/webapp/company-branding-app/components/modal/ExoColorPicker.vue [hunk 8]
```diff
@@ -5,11 +5,14 @@
     color="white"
     width="290px">
     <template v-slot:activator="{ on }">
-      <v-text-field
-        v-model="value"
-        :label="label"
-        readonly
-        v-on="on" />
+      <div style="display: flex;flex-direction: row">
+        <div :style="{ backgroundColor: value}" style="width:47px;height:47px;margin-top:5px;margin-right:10px;cursor: pointer" v-on="on">
```
Comment (azayati) [inline portlet/exoadmin/src/main/webapp/company-branding-app/components/modal/ExoColorPicker.vue]: Avoid using style directly

File: portlet/exoadmin/src/main/webapp/company-branding-app/components/modal/ExoColorPicker.vue [hunk 9]
```diff
@@ -5,11 +5,14 @@
...
+        </div>
+        <div>
+          <p style="font-style: oblique;font-size: 16px;color: #999999;margin-bottom: 7px;margin-top: 3px;">{{ label }}</p>
+          <label style="cursor: pointer" v-on="on">{{ value }}</label>
```
Comment (azayati) [inline portlet/exoadmin/src/main/webapp/company-branding-app/components/modal/ExoColorPicker.vue]: Avoid using style directly

File: portlet/exoadmin/src/main/resources/locale/portlet/exoadmin/Branding_en.properties [hunk 10]
```diff
@@ -20,3 +20,9 @@ style.dark.label=Dark
 style.light.label=Light
 attachment=Drop image here or upload from your computer
 companyName.placeholder=Type your company name here
+branding.select.attachment=Upload manually
+branding.drop.attachment=Drop image here
+branding.or.attachment=or
+branding.message.change.logo=Click here to change logo
+branding.message.remove.logo=Click here to restore the default logo
```
Comment (azayati) [inline portlet/exoadmin/src/main/resources/locale/portlet/exoadmin/Branding_en.properties]:
```suggestion
branding.message.restore.logo=Click here to restore the default logo
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Comment by Rdidmohamedelhadi** [comment]
> See review

I know it's just for fast testing :)

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**